server/jetstream_cluster.go

```diff
@@ -7159,25 +7159,45 @@ func (s *Server) jsClusteredConsumerRequest(ci *ClientInfo, acc *Account, subjec
 		return
 	}
 
+	// Was a consumer name provided?
+	var oname string
+	if isDurableConsumer(cfg) || cfg.Name != _EMPTY_ {
+		if cfg.Name != _EMPTY_ {
+			oname = cfg.Name
+		} else {
+			oname = cfg.Durable
+		}
+	}
+
 	// Check for max consumers here to short circuit if possible.
 	// Start with limit on a stream, but if one is defined at the level of the account
 	// and is lower, use that limit.
-	maxc := sa.Config.MaxConsumers
-	if maxc <= 0 || (selectedLimits.MaxConsumers > 0 && selectedLimits.MaxConsumers < maxc) {
-		maxc = selectedLimits.MaxConsumers
-	}
-	if maxc > 0 {
-		// Don't count DIRECTS.
-		total := 0
-		for _, ca := range sa.consumers {
-			if ca.Config != nil && !ca.Config.Direct {
-				total++
+	if action == ActionCreate || action == ActionCreateOrUpdate {
```

Comment:
For ` ActionCreateOrUpdate`, what if we determine it is an update?

Reply:
The `if oname != "" && cn == oname && sa.consumers[oname] != nil` is checking if we think it's an update, if it is then we skip counting the matching consumer assignment so that we don't hit the `>=` case that returns an error. 

If it turns out not to be a genuine update (i.e. the consumer doesn't exist) then it doesn't matter as there are checks below that catch that case and error out.